lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -623,4 +646,38 @@ class AwsCompileFunctions {
   }
 }
 
+function addFileContentsToHashes(filePath, hashes) {
+  return new BbPromise((cb, reject) => {
+    const readStream = fs.createReadStream(filePath);
+    readStream
+      .on('data', chunk => {
+        hashes.forEach(hash => {
+          hash.write(chunk);
+        });
+      })
+      .on('close', () => {
+        cb();
+      })
+      .on('error', error => {
+        reject(new Error(`Could not add file content to hash: ${error}`));
+      });
+  });
+}
+
+function extractLayerConfigurationsFromFunction(functionProperties, cfTemplate) {
+  const layerConfigurations = [];
+  if (!functionProperties.Layers) return layerConfigurations;
+  functionProperties.Layers.forEach(potentialLocalLayerObject => {
+    if (potentialLocalLayerObject.Ref) {
+      const configuration = cfTemplate.Resources[potentialLocalLayerObject.Ref];
+      layerConfigurations.push({
+        serverlessLayerName: configuration._serverlessLayerName,
```

Comment:
I think we should name property `name`. Firstly in this context `serverlessLayer` prefix is not justified, and when processing it, in one place we assume it's name (https://github.com/serverless/serverless/blob/76bf822475eb37e801e9cce3b533e7cd7bf608ab/lib/plugins/aws/package/compile/functions/index.js#L419) which now resolves to `undefined`

Reply:
Thanks, good catch. I think it must have been `name` at one point, so I've changed it back. I've added the other improvements and merged in this change from master (https://github.com/serverless/serverless/commit/1fceb898d0ea10b00bc6759a5204065c81b560e8) so let me know me if anything looks wrong with that.